src/ex4/server/services/storage.js

```diff
@@ -0,0 +1,72 @@
+import Item from "../db/models/item.js";
+import { sequelize } from "../db/models/index.js";
+
+class StorageService {
+  async getTasks() {
+    return await Item.findAll({
+      attributes: [
+        "id",
+        "itemName",
+        "checked",
+        [
+          sequelize.fn(
+            "DATE_FORMAT",
+            sequelize.col("doneTime"),
+            "%Y-%m-%d %H:%i:%s"
+          ),
+          "doneTime",
+        ],
+      ],
+      raw: true,
+    });
+  }
+
+  async addTask(itemName) {
+    const addedTask = await Item.create({ itemName }, { raw: true });
+    return addedTask.toJSON().id;
+  }
+
+  async deleteTask(id) {
+    return await Item.destroy({ where: { id } });
+  }
+
+  async clearAll() {
+    return await Item.destroy({ truncate: true });
+  }
+
+  getCurrTime() {
+    let now = new Date().getTime();
+    now += 3 * 60 * 60 * 1000; // add 3 hours to match to Israel Time Zone
+    // return new Date(now)
+    return new Date(now);
+  }
+
+  async updateTask(id, updatedKeysValues) {
+    const doneTime = updatedKeysValues?.checked ? this.getCurrTime() : null;
```

Comment:
is `updateTasks` always set task to done? otherwise you have data corruption here as the `doneTime` is change on each update and does not represents the real done time